fe/fe-core/src/test/java/com/starrocks/sql/plan/JoinTest.java

```diff
@@ -1183,56 +1183,56 @@ public void testConstPredicateInRightJoin() throws Exception {
         Assert.assertTrue(explainString.contains("  2:OlapScanNode\n" +
                 "     TABLE: join2\n" +
                 "     PREAGGREGATION: ON\n" +
-                "     PREDICATES: round(2, 0) > 3"));
+                "     PREDICATES: CAST(round(2.0, 0) AS DOUBLE) > 3.0"));
 
         sql = "select * from test.join1 right semi join test.join2 on join1.id = join2.id where round(2.0, 0) > 3.0";
         explainString = getFragmentPlan(sql);
         Assert.assertTrue(explainString.contains("  0:OlapScanNode\n" +
                 "     TABLE: join2\n" +
                 "     PREAGGREGATION: ON\n" +
-                "     PREDICATES: round(2, 0) > 3"));
+                "     PREDICATES: CAST(round(2.0, 0) AS DOUBLE) > 3.0"));
 
         sql = "select * from test.join1 right anti join test.join2 on join1.id = join2.id where round(2.0, 0) > 3.0";
         explainString = getFragmentPlan(sql);
         Assert.assertTrue(explainString.contains("  0:OlapScanNode\n" +
                 "     TABLE: join2\n" +
                 "     PREAGGREGATION: ON\n" +
-                "     PREDICATES: round(2, 0) > 3"));
+                "     PREDICATES: CAST(round(2.0, 0) AS DOUBLE) > 3.0"));
```

Comment:
Why we need a cast double here? but needn't before this PR.

Reply:
Before this pr, `3.0` will be treated as `integer` because the trailing zero is removed and the `round(2.0, 0)` will output decimal type with `scale=0`, and no cast needed here.

After this pr, `3.0` will be treated as `float point` so there need a cast before comparition.

Reply:
The query result maybe change because of type change.